src/OpenTelemetry.Extensions.PersistentStorage.Api/IPersistentBlob.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="IPersistentBlob.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Extensions.PersistentStorage.Api
+{
+    /// <summary>
+    /// Represents a persistent blob.
+    /// </summary>
+    public interface IPersistentBlob
+    {
+        /// <summary>
+        /// Reads the content from the blob.
+        /// </summary>
+        /// <returns>
+        /// The content of the blob if the operation succeeded, otherwise null.
+        /// </returns>
+        /// <remarks>
+        /// This function should never throw exception.
```

Comment:
Who is the target audience for the exception remark? Maybe it should be: "Note to implementers: This function should never throw exception." 🤷

Reply:
Agree abstract class may be nice here especially as a place to put any common logic like the suggestion here https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/221/files/0e1f10b4c24a2058d5c889a3655ca0bfa380c935#r852365170

Reply:
it is intended for implementers - I will update it.